harness-delegate-ng/templates/_helpers.tpl

```diff
@@ -101,3 +101,25 @@ Generate a list of default certificate path with certificate target location
   {{- end }}
   {{- join "," $mount_volumes -}}
 {{- end }}
+
+{{/*
+Define the delegate token name
+*/}}
+{{- define "harness-delegate-ng.delegateToken" -}}
+{{- if .Values.delegateToken }}
+{{- include "harness-delegate-ng.fullname" . }}
+{{- else }}
+{{- .Values.existingDelegateToken | trunc 63 | toString }}
+{{- end }}
+{{- end }}
+
+{{/*
+Define the upgrader delegate token name
+*/}}
+{{- define "harness-delegate-ng.upgraderDelegateToken" -}}
+{{- if .Values.delegateToken }}
+{{- include "harness-delegate-ng.fullname" . }}-upgrader-token
+{{- else }}
+{{- .Values.existingDelegateToken | trunc 63 | toString }}
```

Comment:
I don't think this would work for upgrader. Can you please run upgrader and make sure it succeeds when installed this way.
Note that even though secret value can be the same between delegate and upgrader (though also it doesn't have to be, you could have separate token for delegate and separate for upgrader) upgrader uses a different name for ENV variable (`UPGRADER_TOKEN` vs `DELEGATE_TOKEN` for delegate) containing the secret value. So if you just mount delegate's secret it would have variable named DELEGATE_TOKEN and fail (and vice versa if you mount upgraders secret to delegate). I think you would need two new variables in values.yaml `existingDelegateToken` and `existingUpgraderToken`. 
Also, note that we refer to Upgrader only as Upgrader not DelegateUpgrader (because it's in charge for more than just delegate), so name the variables accordingly (e.g. `harness-delegate-ng.upgraderToken`, `existingUpgraderToken` etc...)